banzai/dbs.py

```diff
@@ -337,10 +337,9 @@ def get_instrument(header, db_address=_DEFAULT_DB):
 
 def get_bpm_filename(instrument_id, ccdsum, db_address=_DEFAULT_DB):
     db_session = get_session(db_address=db_address)
-    bpm_query = db_session.query(CalibrationImage).filter(
-                    CalibrationImage.type == 'BPM',
-                    CalibrationImage.instrument_id == instrument_id,
-                    cast(CalibrationImage.attributes['ccdsum'], String) == type_coerce(ccdsum, JSON))
+    bpm_query = db_session.query(CalibrationImage).filter(CalibrationImage.type == 'BPM',
+                                                          CalibrationImage.instrument_id == instrument_id,
+                                                          CalibrationImage.attributes['ccdsum'] == type_coerce(ccdsum, JSON))
```

Comment:
Are you sure you want to take remove `cast`? From [here](https://docs.sqlalchemy.org/en/latest/core/type_basics.html?highlight=json#sqlalchemy.types.JSON):
>Index operations return an expression object whose type defaults to JSON by default, so that further JSON-oriented instructions may be called upon the result type. Note that there are backend-specific idiosyncracies here, including that the PostgreSQL database does not generally compare a “json” to a “json” structure without type casts. These idiosyncracies can be accommodated in a backend-neutral way by making explicit use of the cast() and type_coerce() constructs. Comparison of specific index elements of a JSON object to other objects works best if the left hand side is CAST to a string and the right hand side is rendered as a JSON string

Reply:
Added back.